src/DateTimeRoutines/DateTimeRoutines.cs

```diff
@@ -396,49 +392,49 @@ public static bool TryParseTime(this string str, DateTimeFormat default_format,
                     return false;
             }
 
-            if (string.Compare(m.Groups["ampm"].Value, "PM", true) == 0 && hour < 12)
+            if (string.Compare(m.Groups["ampm"].Value, "PM", StringComparison.OrdinalIgnoreCase) == 0 && hour < 12)
```

Comment:
According to https://docs.microsoft.com/en-us/dotnet/standard/base-types/best-practices-strings best performance would be to also change to `string.Equals`

It also says `InvariantCulture` comparers should be avoided unless you have a specific reason to use them. I did not know this. Ordinal is the best option for performance.

Reply:
changed